modules/price/src/Resolver/DefaultPriceResolver.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Drupal\commerce_price\Resolver;
+
+use Drupal\commerce\PurchasableEntityInterface;
+use Drupal\commerce_price\Plugin\Field\FieldType\Price;
+
+/**
+ * Returns the price based on the purchasable entity's price field.
+ */
+class DefaultPriceResolver implements PriceResolverInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function resolve(PurchasableEntityInterface $entity, $quantity = 1) {
+    return $this->applies($entity) ? $entity->price->first()->toPrice() : NULL;
+  }
+
+  /**
+   * Determines whether the resolver applies to the given purchasable entity.
+   *
+   * @param \Drupal\commerce\PurchasableEntityInterface $entity
+   *   The purchasable entity.
+   *
+   * @return bool
+   *   TRUE if the resolver applies to the given purchasable entity, FALSE
+   *   otherwise.
+   */
+  public function applies(PurchasableEntityInterface $entity) {
```

Comment:
We need to put applies() on the interface, and have it be before resolve(). Invoked by the chain resolver, instead of called by resolve(). The current solution is ugly for no visible gain